core/src/main/java/org/apache/accumulo/core/util/PeekingIterator.java

```diff
@@ -91,4 +94,39 @@ public boolean hasNext() {
     }
     return top != null;
   }
+
+  /**
+   * Advances the underlying iterator looking for a match, inspecting up to {@code limit} elements
+   * from the iterator. If this method finds a match to the predicate, then it will return true and
+   * will be positioned before the matching element (peek() and next() will return the matching
+   * element). If this method does not find a match because the underlying iterator ended before
+   * {@code limit}, then it will return false and hasNext will also return false. Otherwise, if this
+   * method does not find a match, then it will return false and be positioned before the limit
+   * element (peek() and next() will return the {@code limit} element).
+   *
+   * @param predicate condition that we are looking for, parameter could be null, so the Predicate
+   *        implementation needs to handle this.
+   * @param limit number of times that we should look for a match, parameter should be a positive
```

Comment:
```suggestion
   * @param limit number of times that we should look for a match, parameter must be a positive
```
`should` suggests that a negative value could be passed but is not recommended. However doing that always results in an  `IllegalArgumentException` being thrown from line 118